### DISCUSSION TITLE
No filter.d updates on Ubuntu v20.04?

### DISCUSSION THREAD

**Discussion by TonyGravagno** [fail2ban/fail2ban#3279] [Q&A] [answered]
I noteiced that while the version for fail2ban is 0.11.1, the filter.d files haven't been updated since installation. The files in this repo are more recent. I've seen the wiki docs here about upgrades but nothing to indicate that there is an intentional reason/lag for config file updates.

Can someone suggest a pattern to ensure that we're getting the approved package from the distro provider while still getting the latest .conf files?

I understand that a change to the software may preclude updates to some support files. I also understand that no one here has influence over what is provided by the distro providers. I know I can manually (automate) download and replace .conf files from this repo in addition to getting distro updates. And I know that local settings need to be saved to .local files to avoid an over-write by upstream config changes.

I'm asking why this situation might exist, and if someone knows that it's intentional, I won't try to thwart the decisions of those who know better.

Thanks!

**Comment by sebres** [comment, marked as answer]
1. Although we trying to hold the filters compatible, they could (even in the same branch) become incompatible to previous version of fail2ban, e. g. some filters of 0.10.5 may don't work properly with 0.10.4 (because would expect an update of fail2ban self) - see for example 1c1d2cc435d6e8f1eb4a1b60c935a1385a82e295, the sshd-filter got `<F-MLFGAINED>` tag in regex and this was implemented in fail2ban, so such a filter update would also expect update of fail2ban modules
2. (good news) I don't know any incompatible filter change since 0.11.1 in (0.11 branch) which would be incompatible to 0.11.1. but of course without guaranty. So theoretically you could update all the filters in `/etc/fail2ban/filter.d` from [fail2ban/tree/0.11/config/filter.d](https://github.com/fail2ban/fail2ban/tree/0.11/config/filter.d), of course they may expect some manual patching normally made for your distribution by the maintainers (that are not under our control).
3. Automated filter update service already discussed in #617

And last but not least - FWIW, [wiki :: How to install or upgrade fail2ban manually](https://github.com/fail2ban/fail2ban/wiki/How-to-install-or-upgrade-fail2ban-manually)
